libc/runtime/zipos-open.c

```diff
@@ -79,13 +83,19 @@ static void *__zipos_mmap_space(size_t mapsize) {
 }
 
 struct ZiposHandle *__zipos_keep(struct ZiposHandle *h) {
-  atomic_fetch_add_explicit(&h->refs, 1, memory_order_relaxed);
+  int refs = atomic_fetch_add_explicit(&h->refs, 1, memory_order_relaxed);
+  if (VERY_UNLIKELY(refs > MAX_REFS)) {
```

Comment:
I recommend `unassert(!VERY_UNLIKELY(refs > MAX_REFS));` for core library stuff like this, since it uses `__builtin_trap()` which is nicer and lighter weight than `abort()` which leads to hairier backtraces. There's also `npassert()` which guarantees the branch is never DCE'd in `NDEBUG` mode.